src/Build/Evaluation/Expander.cs

```diff
@@ -3020,75 +3206,150 @@ internal static void AnyHaveMetadataValue(IElementLocation elementLocation, stri
                         transformedItems.Add(new KeyValuePair<string, S>("false", null));
                     }
                 }
-            }
-
-            /// <summary>
-            /// A functor that returns the value of the metadata in the match
-            /// that is on the item it was created with.
-            /// </summary>
-            private class MetadataMatchEvaluator
```

Comment:
it seems to be conflicting with https://github.com/dotnet/msbuild/pull/12002

cc: @Erarndt